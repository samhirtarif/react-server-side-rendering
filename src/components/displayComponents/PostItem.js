import React from 'react'
import { Link } from 'react-router-dom'

export default ({id, title, date, numberOfComments}) => (
    <div key={id}>
        <Link to={`/posts/${id}`}>{title}</Link>
        <p>{date} | {numberOfComments} {numberOfComments === 1 ? "comment" : "comments"}</p>
    </div>
)