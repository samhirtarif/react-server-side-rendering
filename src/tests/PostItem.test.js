import React from 'react'
import PostItem from '../components/displayComponents/PostItem'
import Enzyme, { shallow } from 'enzyme'
import toJson from "enzyme-to-json";
import { Link } from 'react-router-dom'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

const dummyPost = {
    id : 1,
    title: "Post 1",
    date: "2020-02-02",
    postPreviewContent: "preview1",
    postContent: "content1",
    numberOfComments: 0,
    imgURL: "url/img1"
}

describe('Post item component', () => {
    beforeEach(() => jest.restoreAllMocks())

    it('component renders correctly', async () => {
        // jest.mock('react-router-dom', () => ({
        //   ...jest.requireActual('react-router-dom'),
        //   useParams: jest.fn().mockReturnValue({ id: 1 })
        // }));

        const container = shallow(
          <PostItem {...dummyPost} />
        );

        expect(container.find(Link).length).toEqual(1);
        expect(container.find('p').length).toEqual(1);
    });

    
    it('Component is correctly ordered', async () => {
        const wrapper = shallow(<PostItem {...dummyPost} />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
  });