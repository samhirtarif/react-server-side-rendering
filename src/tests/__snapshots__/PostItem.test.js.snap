// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post item component Component is correctly ordered 1`] = `
<div
  key="1"
>
  <Link
    to="/posts/1"
  >
    Post 1
  </Link>
  <p>
    2020-02-02
     | 
    0
     
    comments
  </p>
</div>
`;
